packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/queryExtender/queryExtender.ts

```diff
@@ -0,0 +1,99 @@
+import type { ModeledDataStream } from '../modeledDataStreamQueryEditor/modeledDataStreamExplorer/types';
+import type { UnmodeledDataStream } from '../unmodeledDataStreamExplorer/types';
+import type { IoTSiteWiseDataStreamQuery } from '~/types';
+
+type Query = IoTSiteWiseDataStreamQuery;
+
+export class QueryExtender {
+  readonly #currentQuery: Query;
+
+  constructor(currentQuery: Query = {}) {
+    this.#currentQuery = currentQuery;
+  }
+
+  public extendAssetQueries(modeledDataStreams: ModeledDataStream[]): Query {
+    const currentAssetQueries = this.#currentQuery.assets ?? [];
+    const newAssetQueries = this.#createAssetQueries(modeledDataStreams);
+    const dedupedAssetQueries = this.#dedupeAssetQueries([...currentAssetQueries, ...newAssetQueries]);
+    const extendedQuery = {
+      ...this.#currentQuery,
+      assets: dedupedAssetQueries,
+    };
+
+    return extendedQuery;
+  }
+
+  #createAssetQueries(modeledDataStreams: ModeledDataStream[]): NonNullable<Query['assets']> {
+    const assetQueriesWithProperties = modeledDataStreams
+      .filter(this.#isModeledDataStream)
+      .map<NonNullable<Query['assets']>[number]>(({ assetId, propertyId: propertyId }) => ({
+        assetId,
+        properties: [{ propertyId }],
+      }));
+
+    return assetQueriesWithProperties;
```

Comment:
nit: you do not need this variable

Reply:
I typically do this for two reasons: 1. To make my data more descriptive. Sometimes it helps. Sometimes it doesn't. 2. To make it easier to debug with console.log or a debugger. I can drop a console.log without changing the code. When I use the debugger, I can reference the variable name.

Reply:
true, thats what I do as well when debugging, I guess its nice to follow one pattern but I can see this is more helpful to read the code